flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/AbstractDeclarativeSlotPoolBridgeTest.java

```diff
@@ -51,37 +52,53 @@ abstract class AbstractDeclarativeSlotPoolBridgeTest {
+    @Parameter(2)
+    boolean slotBatchAllocatable;
+
+    @Parameters(
+            name =
+                    "requestSlotMatchingStrategy: {0}, slotRequestMaxInterval: {1}, slotBatchAllocatable: {2}")
     private static Collection<Object[]> data() {
         return Arrays.asList(
-                new Object[] {SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO},
-                new Object[] {SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ofMillis(50)},
+                new Object[] {SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO, false},
+                new Object[] {SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO, true},
+                new Object[] {
+                    SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ofMillis(20), false
+                },
+                new Object[] {
+                    SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ofMillis(20), true
+                },
+                new Object[] {
+                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO, false
+                },
                 new Object[] {
-                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO
+                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO, true
                 },
                 new Object[] {
-                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE, Duration.ofMillis(50)
+                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE,
+                    Duration.ofMillis(20),
+                    false
+                },
+                new Object[] {
+                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE,
+                    Duration.ofMillis(20),
+                    true
                 });
     }
 
     @Nonnull
     DeclarativeSlotPoolBridge createDeclarativeSlotPoolBridge(
-            DeclarativeSlotPoolFactory declarativeSlotPoolFactory,
-            RequestSlotMatchingStrategy requestSlotMatchingStrategy,
-            Duration slotRequestMaxInterval) {
+            DeclarativeSlotPoolFactory declarativeSlotPoolFactory) {
         return createDeclarativeSlotPoolBridge(
-                declarativeSlotPoolFactory,
-                requestSlotMatchingStrategy,
-                slotRequestMaxInterval,
-                componentMainThreadExecutor);
+                declarativeSlotPoolFactory, componentMainThreadExecutor, null);
     }
 
     @Nonnull
     DeclarativeSlotPoolBridge createDeclarativeSlotPoolBridge(
             DeclarativeSlotPoolFactory declarativeSlotPoolFactory,
-            RequestSlotMatchingStrategy requestSlotMatchingStrategy,
-            Duration slotRequestMaxInterval,
-            ComponentMainThreadExecutor mainThreadExecutor) {
+            ComponentMainThreadExecutor componentMainThreadExecutor,
+            DeclarativeSlotPoolBridgeResourceDeclarationTest.RequirementListener
```

Comment:
Is it possible that we don't pass `RequirementListener` here?

Alternative solution is : All tests of DeclarativeSlotPoolBridgeResourceDeclarationTest could call `requirementListener#tryWaitSlotRequestIsDone` directly.

In general, the parent class doesn't depends on the class inside of children class.